python-lib/tagger_formatter.py

```diff
@@ -0,0 +1,443 @@
+                    "tag": self.tagger.keyword_to_tag[keyword.text],
+                }
+                for keyword in match
+            ]
+            self._update_output_df(match, values, row)
+
+    def _get_tags_row_per_keyword_category(self, document: List, row: pd.Series):
+        """
+        Called by _row_per_keyword_category
+        Creates the list of new rows with infos about the tags and gives it to _update_output_df function
+        """
+        tag_rows = []
+        for sentence in document:
+            tag_rows = [
+                {
+                    "tag_keyword": keyword.text,
+                    "tag_sentence": sentence.text,
+                    "tag_category": keyword.label_,
+                    "tag": self.tagger.keyword_to_tag[keyword.text],
+                }
+                for keyword in sentence.ents
+            ]
+            self._update_output_df(tag_rows, tag_rows, row)
+
+    def _update_output_df(self, match: List, values: List[dict], row: pd.Series):
+        """
+        Called by _get_tags_row_per_keyword_category and _get_tags_row_per_keyword
+        Appends:
+        -row with infos about the founded tags to output_df
+        -duplicated initial row from the Document dataframe(text_df) to df.duplicated_lines
+        """
+        if match:
+            self.output_df = self.output_df.append(values, ignore_index=True)
+            self.df_duplicated_lines = self.df_duplicated_lines.append(
+                [row for i in range(len(values))]
+            )
+            self.matches = True
+
+    def _merge_df_columns(self):
+        """
+        Called by _row_per_document
+        insert columns tag_sentences and tag_keywords
+        returns the complete output dataframe
+        """
+        self.output_df.insert(
+            0, self.output_dataset_columns[1], self.tag_sentences, True
+        )
+        self.output_df.insert(
+            1, self.output_dataset_columns[0], self.tag_keywords, True
+        )
+        self.output_df = pd.concat(
+            [
+                self.tagger.text_df.drop(
+                    columns=[self.tagger.splitted_sentences_column]
+                ),
+                self.output_df,
+            ],
+            axis=1,
+        )
+        return self._arrange_columns_order(
+            self.output_df,
+            self.output_dataset_columns + [self.tagger.text_column],
+        )
+
+    def _arrange_columns_order(self, df: pd.DataFrame, columns: List[AnyStr]):
+        """Put columns in the right order in the Output dataframe"""
+        for column in columns:
+            df.set_index(column, inplace=True)
+            df.reset_index(inplace=True)
+        return df
+
+    def _get_output_dataset_columns(self):
+        """Returns the list of additional columns for the output dataframe"""
+        if self.tagger.category_column:
+            if self.tagger.mode == OutputFormat.ONE_ROW_PER_DOC_JSON.value:
```

Comment:
It's odd to have something related to OutputFormat related to the `self.tagger`. It would be better to define it in the context of the TaggerFormatter